doc/source/programs/gdal_vector_sql.rst

```diff
@@ -0,0 +1,105 @@
+
+    Positional arguments:
+      -i, --input <INPUT>                                  Input vector dataset [required]
+      -o, --output <OUTPUT>                                Output vector dataset [required]
+      --sql <statement>|@<filename>                        SQL statement(s) [may be repeated] [required]
+
+    Common Options:
+      -h, --help                                           Display help message and exit
+      --version                                            Display GDAL version and exit
+      --json-usage                                         Display usage as JSON document and exit
+      --drivers                                            Display driver list as JSON document and exit
+      --config <KEY>=<VALUE>                               Configuration option [may be repeated]
+      --progress                                           Display progress bar
+
+    Options:
+      -l, --layer, --input-layer <INPUT-LAYER>             Input layer name(s) [may be repeated]
+      -f, --of, --format, --output-format <OUTPUT-FORMAT>  Output format
+      --co, --creation-option <KEY>=<VALUE>                Creation option [may be repeated]
+      --lco, --layer-creation-option <KEY>=<VALUE>         Layer creation option [may be repeated]
+      --overwrite                                          Whether overwriting existing output is allowed
+      --update                                             Whether to open existing dataset in update mode
+      --overwrite-layer                                    Whether overwriting existing layer is allowed
+      --append                                             Whether appending to existing layer is allowed
+      --output-layer <OUTPUT-LAYER>                        Output layer name(s) [may be repeated]
+      --dialect <DIALECT>                                  SQL dialect (e.g. OGRSQL, SQLITE)
+
+    Advanced Options:
+      --if, --input-format <INPUT-FORMAT>                  Input formats [may be repeated]
+      --oo, --open-option <KEY=VALUE>                      Open options [may be repeated]
+
+
+Description
+-----------
+
+:program:`gdal vector sql` returns one or several layers evaluated from
+SQL statements.
+
+Standard options
+++++++++++++++++
+
+.. include:: gdal_options/of_vector.rst
+
+.. include:: gdal_options/co_vector.rst
+
+.. include:: gdal_options/overwrite.rst
+
+.. option:: --sql <sql_statement>|@<filename>
+
+    SQL statement to execute that returns a table/layer (typically a SELECT
+    statement)
+
+    Can be repeated to generated multiple output layers.
+
+.. option:: --dialect <DIALECT>
+
+    SQL dialect. In some cases can be used to use the (unoptimized) :ref:`ogr_sql_dialect` instead
+    of the native SQL of an RDBMS by passing the ``OGRSQL`` dialect value.
+    The :ref:`sql_sqlite_dialect` dialect can be chosen with the ``SQLITE``
+    and ``INDIRECT_SQLITE`` dialect values, and this can be used with any datasource.
+
```

Comment:
My attempt on this:

SQL dialect. By default the native SQL of an RDBMS is used. If a datasource does not support SQL natively, the default is to use the OGRSQL dialect. OGRSQL can by used with any datasource. The SQLITE dialect can be chosen with the ``SQLITE`` and ``INDIRECT_SQLITE`` dialect values, and they can also be used with any datasource. Overriding the default dialect may be beneficial because the capabilities of the SQL dialects vary. What SQL dialects a driver supports can be checked with "gdal vector info".
```
gdal vector info --format "PostgreSQL"
Supported SQL dialects: NATIVE OGRSQL SQLITE
gdal vector info --format "ESRI Shapefile"
Supported SQL dialects: OGRSQL SQLITE

```

Reply:
done. thanks